sympy/solvers/solveset.py

```diff
@@ -1530,8 +1530,12 @@ def _solve_exponential(lhs, rhs, symbol, domain):
     a_term = a.as_independent(symbol)[1]
     b_term = b.as_independent(symbol)[1]
 
-    a_base, a_exp = a_term.base, a_term.exp
-    b_base, b_exp = b_term.base, b_term.exp
+    try:
+        a_base, a_exp = a_term.base, a_term.exp
+        b_base, b_exp = b_term.base, b_term.exp
+    except AttributeError:
```

Comment:
We shouldn't catch AttributeError. I already removed most occurrences of `except AttributeError` from the codebase. The question is why is the AttributeError being raised? Is it a bug in `as_independent`? Or something else?